src/electionguard/guardian.py

```diff
@@ -154,7 +154,13 @@ def __init__(
         sequence_order: int,
```

Comment:
`id` and`sequence_order` are no longer necessary. The ElectionKeyPair includes both these values.

Reply:
Except that ElectionKeyPair is optional.  The vast majority of instantiations of Guardian use `from_context_info` and thus do not provide a key pair.

Reply:
Incorrect. The ElectionKeyPair is never optional. A guardian cannot exist without one. The majority of the existing ones are using the constructor internally. This merely makes it more obvious by pushing it into a specific constructor.

Reply:
You're overcomplicating it.

```
def __init__(
      self,
      key_pair: ElectionKeyPair,
      ceremony_details: CeremonyDetails
) -> None:
      self.id = key_pair.id
      self.sequence_order = key_pair.sequence_order
      self.ceremony_details = ceremony_details

# Applies to all existing cases as they currently are implemented.  
@classmethod
def from_nonce(
        cls, 
        id: str,
        sequence_order: int,
        number_of_guardians: int,
        quorum: int,
        nonce_seed: Optional[ElementModQ] = None):
        key_pair = generate_election_key_pair(
            self.id, self.sequence_order, self.ceremony_details.quorum, nonce
        )
        cls(key_pair, CeremonyDetails(number_of_guardians, quorum))
```

Reply:
I just wrote up an example. It even is able to remove the lint skip. 
https://github.com/microsoft/electionguard-python/tree/feature/355-gaurdian-reconstruction-example